### PR TITLE
Add GitHub Actions to  `gofmt -s -w` all `*.go` files

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds GitHub Actions to  `gofmt -s -w` all `*.go` files,
if any `*.go` files are touched on each commit.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: .github/workflows/go.fmt.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        run: |
+          git config user.name "coredns-auto-go-mod-fmt[bot]"
+          git config user.email "coredns-auto-go-mod-fmt[bot]@users.noreply.github.com"
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+      -
+        name: Commit and push changes
+        run: |
+          git add .
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
+            git commit -m 'auto go mod fmt'
```
Comment (miekg) [inline .github/workflows/go.fmt.yml]:
```suggestion
            git commit -m 'auto fmt'
```

**Comment by miekg** [comment]
change the presubmit whitespace test also to not test go files?

**Comment by yongtang** [comment]
@miekg Created another PR #4269 for removing trailing whitespaces. Want to split out `trailing whitespaces` and `gofmt` into two GitHub Actions (to make sure they are not interference each other.

File: .github/workflows/go.fmt.yml (hunk 1 above)
Comment (yongtang) [reply .github/workflows/go.fmt.yml]: Thanks @miekg, the PR has been updated.

**Pull request merged**